File: game_part/src/com/game/part/tmpl/type/XlsxBool.java
package com.game.part.tmpl.type;

import com.game.part.tmpl.XSSFRowReadStream;

/**
 * Excel Bool 字段
 * 
 * @author hjj2019
 * @since 2015/2/23
 * 
 */
public class XlsxBool extends BasicTypeCol<Boolean> {
	/**
	 * 类默认构造器
	 * 
	 */
	public XlsxBool() {
		super();
	}

	/**
	 * 类参数构造器
	 * 
	 * @param nullable
	 * 
	 */
	public XlsxBool(boolean nullable) {
		super(nullable);
	}

	/**
	 * 类参数构造器
	 * 
	 * @param nullable
	 * @param defaultVal
	 * 
	 */
	public XlsxBool(boolean nullable, boolean defaultVal) {
		super(nullable, defaultVal);
	}

	@Override
	protected void readImpl(XSSFRowReadStream stream) {
		if (stream != null) {
			super.setObjVal(stream.readBool());
		}
	}

	/**
	 * objVal 不能为空, 但如果真为空值, 则自动创建
	 * 
	 * @param objVal
	 * @return
	 * 
	 */
	public static XlsxBool ifNullThenCreate(XlsxBool objVal) {
		if (objVal == null) {
			// 创建对象
			objVal = new XlsxBool();
		}

		return objVal;
	}
}

File: client_wpf/Xgame.GamePart/Tmpl/Type/XlsxBool.cs
using System;

namespace Xgame.GamePart.Tmpl.Type
{
    /// <summary>
    /// Xlsx 布尔型
    /// </summary>
    public class XlsxBool : PrimitiveTypeCol<bool>
    {
        #region 类构造器
        /// <summary>
        /// 类默认构造器
        /// </summary>
        public XlsxBool()
        {
        }

        /// <summary>
        /// 类参数构造器
        /// </summary>
        /// <param name="val"></param>
        public XlsxBool(bool val)
        {
            this.Val = val;
        }
        #endregion

        // @Override
        protected override void ReadImpl(XlsxRowReadStream stream)
        {
            this.Val = stream.ReadBool();
        }
    }
}
